src/replica/test/replica_disk_test.cpp

```diff
@@ -176,42 +123,27 @@ TEST_F(replica_disk_test, on_query_disk_info_all_app)
 
 TEST_F(replica_disk_test, on_query_disk_info_app_not_existed)
 {
-    // test app_id not existed
-    // create fake request
-    dsn::message_ptr fake_request = dsn::message_ex::create_request(RPC_QUERY_DISK_INFO);
-    query_disk_info_request tmp_request;
-    ::dsn::marshall(fake_request, tmp_request);
-
-    // get received request and query disk info
-    dsn::message_ex *recvd_request = fake_request->copy(true, true);
-    auto rpc =
-        rpc_holder<query_disk_info_request, query_disk_info_response>::auto_reply(recvd_request);
-    query_disk_info_request &request = const_cast<query_disk_info_request &>(rpc.request());
+    query_disk_info_request &request =
+        const_cast<query_disk_info_request &>(fake_query_disk_rpc.request());
     request.app_name = "not_existed_app";
-    stub->on_query_disk_info(rpc);
-
-    ASSERT_EQ(rpc.response().err, ERR_OBJECT_NOT_FOUND);
+    stub->on_query_disk_info(fake_query_disk_rpc);
+    ASSERT_EQ(fake_query_disk_rpc.response().err, ERR_OBJECT_NOT_FOUND);
 }
 
 TEST_F(replica_disk_test, on_query_disk_info_one_app)
 {
-    // test app_name = "disk_test_1"
-    // create fake request
-    dsn::message_ptr fake_request = dsn::message_ex::create_request(RPC_QUERY_DISK_INFO);
-    query_disk_info_request tmp_request;
-    ::dsn::marshall(fake_request, tmp_request);
-
-    // get received request and query disk info
-    dsn::message_ex *recvd_request = fake_request->copy(true, true);
-    auto rpc =
-        rpc_holder<query_disk_info_request, query_disk_info_response>::auto_reply(recvd_request);
-    query_disk_info_request &request = const_cast<query_disk_info_request &>(rpc.request());
+    query_disk_info_request &request =
+        const_cast<query_disk_info_request &>(fake_query_disk_rpc.request());
```

Comment:
```suggestion
    query_disk_info_request &request = *fake_query_disk_rpc.mutable_request();
```

Use **mutable_request** instead. `const_cast` is not a good practice in most cases.